File: Tutorial4-8LayerImplementation2/src/main/java/com/test/tutorial4_8layerimplementation2/MyApplication.java
package com.test.tutorial4_8layerimplementation2;


import com.test.tutorial4_8layerimplementation2.di.AppComponent;
import com.test.tutorial4_8layerimplementation2.di.DaggerAppComponent;
import com.test.tutorial4_8layerlibrary.BaseApplication;

import javax.inject.Inject;

import dagger.android.AndroidInjector;
import dagger.android.support.DaggerApplication;

public class MyApplication extends BaseApplication {

    @Inject
    BaseApplication baseApplication;

    @Override
    public void onCreate() {
        super.onCreate();
    }

    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
        AppComponent appComponent = DaggerAppComponent.builder().application(this).build();
        appComponent.inject(this);
        return appComponent;
    }

}

File: Tutorial5-4Dagger-Android-MultipleActivitiesAndFragments/src/main/java/com/smarttoolfactory/tutorial5_4dagger_android_multipleactivitiesandfragments/MyApplication.kt
package com.smarttoolfactory.tutorial5_4dagger_android_multipleactivitiesandfragments

import com.smarttoolfactory.tutorial5_4dagger_android_multipleactivitiesandfragments.di.AppComponent
import com.smarttoolfactory.tutorial5_4dagger_android_multipleactivitiesandfragments.di.DaggerAppComponent
import dagger.android.AndroidInjector
import dagger.android.support.DaggerApplication

class MyApplication : DaggerApplication() {
    override fun applicationInjector(): AndroidInjector<out DaggerApplication> {
        val appComponent: AppComponent = DaggerAppComponent
                .builder()
                .application(this).build()
        appComponent.inject(this)
        return appComponent
    }
}